resources/metadata/kanikoExecute.yaml

```
@@ -119,6 +165,10 @@ spec:
         params:
           - name: container/registryUrl
           - name: container/imageNameTag
+          - name: container/imageNames
+            type: "[]string"
+          - name: container/imageNameTags
+            type: "[]string"
```

I just got reminded of https://github.com/SAP/jenkins-library/issues/2172 and would like to propose a small addition.

```suggestion
          - name: container/imageNames
            type: "[]string"
          - name: container/imageNameTags
            type: "[]string"
          - name: container/imageNameDigests
            type: "[]string"
```

The `container/imageNameDigests` would contain the full image reference using digest format, i.e. `<repo>/<image>@sha256:146dfc...`. This seems to most closely align with the current CPE content.

cc @phil9909 @c0d1ngm0nk3y @pbusko @kulhadia @modulo11 This should be considered for the consumption side within `kubernetesDeploy` as well.

---

@OliverNocon ^^

WDYT?

---

if we see that we need digests later on we can for sure add them. So far no logic exists to fill those.

---

Which might be the point in time to add the logic to fill those, no? Maybe not blocking this PR but then as a follow-up.

It's a not so rare practice in the container ecosystem and it's required to fullfill the weak check of CJ#4 I think.

One could argue that this should be delayed until the image is used for deployment though, and do a `dss` check and digest retrieval in `kubernetesDeploy`.